ci/src/main/scala/org/typelevel/sbt/TypelevelCiPlugin.scala

```diff
@@ -125,7 +132,25 @@ object TypelevelCiPlugin extends AutoPlugin {
 
       style ++ test ++ scalafix ++ mima ++ doc
     },
-    githubWorkflowJavaVersions := Seq(JavaSpec.temurin("8"))
+    githubWorkflowJavaVersions := Seq(JavaSpec.temurin("8")),
+    githubWorkflowAddedJobs ++= {
+      val dependencySubmission =
+        if (tlCiDependencyGraphJob.value)
+          List(
+            WorkflowJob(
+              "dependency-submission",
+              "Submit Dependencies",
+              scalas = List(scalaVersion.value),
+              javas = List(githubWorkflowJavaVersions.value.head),
+              steps = githubWorkflowJobSetup.value.toList :+
+                WorkflowStep
+                  .DependencySubmission
+                  .withCond(Some("github.event_name != 'pull_request'"))
```

Comment:
So I'm a bit confused if we want to enable it for PRs or not. This page seems to suggest that if a PR updates dependencies that it can be reviewed, so it really seems like it would need this to run for that PR.

https://docs.github.com/en/code-security/supply-chain-security/understanding-your-software-supply-chain/about-dependency-review

Reply:
The API submissions are not yet integrated into Dependency Review or Organization Insights, but they do trigger Dependabot alerts, even for transitive dependencies (which the POM integration does not!).

Reply:
I did experiment with dependency review in https://github.com/typelevel/sbt-typelevel/pull/342 but the problem is that PRs from external repos don't have permission to submit the dependencies, so I'm not entirely sure how it would work.

Reply:
I just introduced a vulnerability in a PR, and nothing flagged it until I merged.  I think push for now, and pull once GitHub better integrates their new toys.